### PR TITLE
Add GCP deployer

### PR DESCRIPTION
**Pull request by rpasricha**
#73 

- Update Append builder to add command, user, and environment variables to the image manifest directly.
- Add GCPJob deployment option.

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/fairing/86)

### CODE REVIEW

File: fairing/deployers/gcp/gcp.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+        auth = subprocess.run(['gcloud', 'auth', 'print-access-token'],
+                              stdout=subprocess.PIPE).stdout[:-1].decode('utf-8')
+        headers = {
+            'Content-Type': 'application/json',
+            'Authorization': 'Bearer {}'.format(auth)
+        }
+
+        try:
+            r = requests.post('https://ml.googleapis.com/v1/projects/{}/jobs'.format(self._project_id),
```
Comment (r2d4) [inline fairing/deployers/gcp/gcp.py]: Why aren't you using the GCP python library?

File: fairing/builders/append/append.py [hunk 2]
```diff
@@ -73,7 +74,10 @@ def _build(self, transport, src):
         creds = docker_creds.DefaultKeychain.Resolve(src)
         with v2_2_image.FromRegistry(src, creds, transport) as src_image:
             with open(self.context_file, 'rb') as f:
-                new_img = append.Layer(src_image, f.read())
+                new_img = append.Layer(src_image, f.read(),
```
Comment (r2d4) [inline fairing/builders/append/append.py]: Can you add this to the dockerfile generator as well? Or open up an issue to do it?

File: fairing/config.py [hunk 3]
```diff
@@ -36,7 +37,8 @@
 
 deployer_map = {
     'job': Job,
-    'tfjob': TfJob
+    'tfjob': TfJob,
+    'gcp': GCPJob
```
Comment (r2d4) [inline fairing/config.py]: GCP the right name? Or CMLE?

File: fairing/deployers/gcp/gcp.py (hunk 1 above)
Comment (rpasricha) [reply fairing/deployers/gcp/gcp.py]: Good catch, updated to use the Python client library

File: fairing/config.py (hunk 3 above)
Comment (rpasricha) [reply fairing/config.py]: I used GCP to keep it general as possible as a managed deployment option. What are your thoughts?

File: fairing/builders/append/append.py (hunk 2 above)
Comment (rpasricha) [reply fairing/builders/append/append.py]: Sure, I'll create an issue for this

Comment (rpasricha) [reply fairing/builders/append/append.py]: Actually, looks like one already exists -- #65

File: fairing/config.py (hunk 3 above)
Comment (karthikv2k) [reply fairing/config.py]: LGTM

**Pull request merged**